documentation/ProjectReference-Protocol.md

```diff
@@ -0,0 +1,73 @@
+
+## Who this document is for
+
+This document describes that process, including what is required of a project to be referenceable through a `ProjectReference`. It is intended for for MSBuild SDK maintainers, and those who have created a completely custom project type that needs to interoperate with other projects. It may also be of interest if you'd like to see the implementation details of references. Understanding the details should not be necessary to _use_ `ProjectReferences` in your project.
+
+## Targets related to consuming a reference
+
+The bulk of the work of transforming `ProjectReference` items into something suitable to feed the compiler is done by tasks listed in the `ResolveReferencesDependsOn` property defined in `Microsoft.Common.CurrentVersion.targets`.
+
+There are empty hooks in the default targets for
+
+* `BeforeResolveReferences`—run before the primary work of resolving references.
+* `AfterResolveReferences`—run after the primary work of resolving references.
+
+`AssignProjectConfiguration` runs when building in a solution context, and ensures that the right `Configuration` and `Platform` are assigned to each reference. For example, if a solution specifies (using the Solution Build Manager) that for a given solution configuration, a project should always be built `Release`, that is applied inside MSBuild in this target.
+
+`PrepareProjectReferences` then runs, ensuring that each referenced project exists (creating the item `@(_MSBuildProjectReferenceExistent)`) and asking it for the closest matching `TargetFramework` to build.
+
+`ResolveProjectReferences` does the bulk of the work, building the referenced projects and collecting their outputs.
+
+After the compiler is invoked, `GetCopyToOutputDirectoryItems` pulls child-project outputs into the current project to be copied to its output directory.
+
+When `Clean`ing the output of a project, `CleanReferencedProjects` ensures that referenced projects also clean.
+
+## Targets required to be referenceable
+
+These targets should exist in a project to be compatible with the common targets' `ProjectReference`. Some are called only conditionally.
+
+These targets are all defined in `Microsoft.Common.targets` and are defined in Microsoft SDKs. You should only have to implement them yourself if you require custom behavior or are authoring a project that doesn't import the common targets.
+
+If implementing a project with an “outer” (determine what properties to pass to the real build) and “inner” (fully specified) build, only `GetTargetFrameworkProperties` is required in the “outer” build. The other targets listed can be “inner” build only.
+
+<!--* `BuildGenerateSources` is run
+  * **Conditions**: only if `'$(BuildPassReferences)' == 'true'`.
+  * Rare in managed projects but common in C++ scenarios (for example, using IDL to generate header code).-->
+* `GetTargetFrameworkProperties` determines what properties should be passed to the main get-build-output target.
```

Comment:
It's `GetTargetPath` in VS, default target by default otherwise, but the default target can be overridden by the consumer by setting the metadata `Targets` on the `ProjectReference`.